File: backend/controllers/authController.js
const bcrypt = require('bcryptjs');
const jwt = require('jsonwebtoken');
const { createUser, getUserByEmail } = require('../models/userModel');

const register = async (req, res) => {
  try {
    const { name, email, password, role } = req.body;
    if (!name || !email || !password) {
      return res.status(400).json({ message: 'Missing required fields' });
    }

    const existingUser = await getUserByEmail(email);
    if (existingUser) {
      return res.status(409).json({ message: 'Email already registered' });
    }

    const salt = await bcrypt.genSalt(10);
    const hashedPassword = await bcrypt.hash(password, salt);
    const userRole = role === 'employer' ? 'employer' : 'seeker';

    const userId = await createUser(name, email, hashedPassword, userRole);
    
    const token = jwt.sign(
      { user_id: userId, role: userRole },
      process.env.JWT_SECRET || 'super-secret-key-change-me',
      { expiresIn: '1d' }
    );

    res.status(201).json({
      message: 'User created successfully',
      token,
      user: { id: userId, name, email, role: userRole }
    });
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
};

const login = async (req, res) => {
  try {
    const { email, password } = req.body;
    if (!email || !password) {
      return res.status(400).json({ message: 'Missing required fields' });
    }

    const user = await getUserByEmail(email);
    if (!user) {
      return res.status(401).json({ message: 'Invalid credentials' });
    }

    const isMatch = await bcrypt.compare(password, user.password_hash);
    if (!isMatch) {
      return res.status(401).json({ message: 'Invalid credentials' });
    }

    const token = jwt.sign(
      { user_id: user.id, role: user.role },
      process.env.JWT_SECRET || 'super-secret-key-change-me',
      { expiresIn: '1d' }
    );

    res.status(200).json({
      message: 'Login successful',
      token,
      user: { id: user.id, name: user.name, email: user.email, role: user.role }
    });
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
};

module.exports = {
  register,
  login
};

File: backend/controllers/auth_controller.py
from flask import request, jsonify
import bcrypt
from models.user import create_user, get_user_by_email
from utils.auth import generate_token

def register():
    data = request.get_json()
    if not data or not data.get('name') or not data.get('email') or not data.get('password'):
        return jsonify({'message': 'Missing required fields'}), 400
        
    existing_user = get_user_by_email(data['email'])
    if existing_user:
        return jsonify({'message': 'Email already registered'}), 409
        
    hashed_password = bcrypt.hashpw(data['password'].encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
    role = data.get('role', 'seeker')
    
    if role not in ['seeker', 'employer']:
        return jsonify({'message': 'Invalid role'}), 400
        
    try:
        user_id = create_user(data['name'], data['email'], hashed_password, role)
        token = generate_token(user_id, role)
        return jsonify({'message': 'User created successfully', 'token': token, 'user': {'id': user_id, 'name': data['name'], 'email': data['email'], 'role': role}}), 201
    except Exception as e:
        return jsonify({'message': str(e)}), 500

def login():
    data = request.get_json()
    if not data or not data.get('email') or not data.get('password'):
        return jsonify({'message': 'Missing required fields'}), 400
        
    user = get_user_by_email(data['email'])
    if not user:
        return jsonify({'message': 'Invalid credentials'}), 401
        
    if bcrypt.checkpw(data['password'].encode('utf-8'), user['password_hash'].encode('utf-8')):
        token = generate_token(user['id'], user['role'])
        return jsonify({
            'message': 'Login successful',
            'token': token,
            'user': {
                'id': user['id'],
                'name': user['name'],
                'email': user['email'],
                'role': user['role']
            }
        }), 200
        
    return jsonify({'message': 'Invalid credentials'}), 401
